src/makeInfinityAdder.js

```diff
@@ -6,7 +6,23 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  let totalSum = 0;
+
+  function infinitySum(currentNum) {
+    if (currentNum === undefined) {
+      const currentSum = totalSum;
+
+      totalSum = 0;
+
+      return currentSum;
+    }
+
+    totalSum += currentNum;
+
+    return infinitySum;
+  }
+
+  return infinitySum;
```

Comment:
`makeInfinityAdder` gonna make `infinityAdder` )
```suggestion
  return infinityAdder;
```